sch_fq.c

```diff
@@ -321,6 +467,33 @@ static struct fq_flow *fq_classify(struct sk_buff *skb, struct fq_sched_data *q)
 				     f->socket_hash != sk->sk_hash)) {
 				f->credit = q->initial_quantum;
 				f->socket_hash = sk->sk_hash;
+				//printk("flow hash in rb tree value of each flow is  : %u \n ",f->socket_hash );
+				if((pFlowid[0] == -1) && (pFlowid[1] ==-1) )
+				{
+				pFlowid[0] = f->socket_hash;
+				printk("flow pflowid 0 hash in rb tree value of each flow is  : %u \n ",pFlowid[0] );
+				/*if(pFlowid[0] == 0)
+					{
+				resetFlowid(pFlowid);
+					}*/
+				}
+				
+				if((pFlowid[0] != -1 ) && (pFlowid[1] == -1) )
+				{
+				
+				int lVal = valuePresentinArray(f->socket_hash,pFlowid);
+				
+				if(pFlowid[0] != f->socket_hash)
+				pFlowid[1] = f->socket_hash;
+				
+				printk("flow pflowid 1 hash in rb tree value of each flow is  : %u \n ",pFlowid[1] );
+				
+				/*if((pFlowid[1] == 0) )
+					{
+				resetFlowid(pFlowid);
+					}*/
+				}
+
```

Comment:
So far from this function it's still unclear to me why flows with different `socket_hash` would still be duplicated within the `pFlowId` array. 
We must seriously consider unit testing the function `valuePresentInArray` to ensure it does what we think.